### ISSUE TITLE
[Feature request] TODO: Identify bash case modification

### ISSUE THREAD

**Issue by PeterDaveHello** [koalaman/shellcheck#1642]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

TEXT="this is DEMO text."

echo "${TEXT^}"
echo "${TEXT^^}"
echo "${TEXT,}"
echo "${TEXT,,}"
```

#### Here's what shellcheck currently says:

> No issues detected!

#### Here's what I wanted or expected to see:

This is the "case modification" feature supported by bash v4.x+, result in:
```
This is DEMO text.
THIS IS DEMO TEXT.
this is DEMO text.
this is demo text.
```

POSIX sh should return `bad substitution` without valid output.

So the expected message will be 
```
In POSIX sh, ${VAR^} is undefined.
In POSIX sh, ${VAR^^} is undefined.
In POSIX sh, ${VAR,} is undefined.
In POSIX sh, ${VAR,,} is undefined.
```

Thank you!

**Comment by e-kwsm** [comment]
[SC3059](https://www.shellcheck.net/wiki/SC3059) covers this.
